tests/data/iterators/lazy_iterator_test.py

```diff
@@ -0,0 +1,79 @@
+# pylint: disable=no-self-use,invalid-name
+from typing import List
+
+from allennlp.common import Params
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data import LazyDataset, Instance, Token, Vocabulary
+from allennlp.data.fields import TextField
+from allennlp.data.iterators import LazyIterator
+from allennlp.data.token_indexers import SingleIdTokenIndexer
+
+class LazyIteratorTestCase(AllenNlpTestCase):
+    def setUp(self):
```

Comment:
Why not just put these methods in `TestLazyIterator`?

Reply:
because I copied the `BasicIterator` tests and that's how they were written 😄

Reply:
Ah, right, those actually do get used in other places, because the other iterators are so similar.